app/actions/get_user_from_omniauth_action.rb

```diff
@@ -0,0 +1,35 @@
+# frozen_string_literal: true
+
+class GetUserFromOmniauthAction < ApplicationAction
+  option :access_token
+
+  private
+
+  def perform_implementation
+    data = access_token.info
+    user = User.find_by(email: data['email'])
+
+    ActiveRecord::Base.transaction do
+      if user.nil?
+        user = create_user(data)
```

Comment:
Repeating the question: can we avoid variable reassignment?

Reply:
The goal and a solution. I don't see both.

Reply:
Please elaborate on what you mean. I don't follow.

Reply:
You mean you can't get to the following?

```ruby
  def perform_implementation
    ActiveRecord::Base.transaction do
      existing_user(access_token.info) || create_user(access_token.info)
    end
  end
```

Can you implement existing_user and create_user now? Would it solve my concern above?